src/java/org/apache/cassandra/db/repair/CassandraValidationIterator.java

```diff
@@ -195,12 +195,19 @@ public CassandraValidationIterator(ColumnFamilyStore cfs, Collection<Range<Token
             if (!isIncremental)
             {
                 // flush first so everyone is validating data that is as similar as possible
-                StorageService.instance.forceKeyspaceFlush(cfs.keyspace.getName(), cfs.name);
+                cfs.forceBlockingFlush(ColumnFamilyStore.FlushReason.REPAIR);
+                // Note: we also flush for incremental repair during the anti-compaction process.
             }
             sstables = getSSTablesToValidate(cfs, ranges, parentId, isIncremental);
         }
 
+        // Persistent memtables will not flush or snapshot to sstables, make an sstable with their data.
+        cfs.writeAndAddMemtableRanges(parentId,
+                                      () -> Collections2.transform(Range.normalize(ranges), Range::makeRowRange),
+                                      sstables);
```

Comment:
I guess there's a general question around how much of the persistent Memtable support we want to add in this Jira when we know we probably won't have one in-tree for a wile (if ever). I'd be fine keeping these things if we could also come up with a very simple toy/test implementation that we can base a smoke test on...